bin/cli.js

```diff
@@ -9,75 +10,96 @@ const yargs = require('yargs');
-    'config': 'Read config form file.',
-    'disable-cpu-throttling': 'Disable CPU throttling',
-    'and more Lighthouse CLI options': 'Docs - https://github.com/GoogleChrome/lighthouse/#lighthouse-cli-options',
+  .wrap(yargs.terminalWidth())
+  .usage('Usage: $0 <url>')
+  .command('url', 'URL to test')
+  .option('json', {
+    'describe': 'JSON output defaults to stdout.',
+    'type': 'boolean',
+    'default': false,
+    'group': 'Output:'
   })
-  .default('disable-cpu-throttling', false)
-  .default('runs', 1)
-
-  // boolean values
-  .boolean([
-    'disable-cpu-throttling'
-  ])
+  .option('output-path', {
+    'describe': 'The file path to output the results',
+    'type': 'string',
+    'default': '',
+    'group': 'Output:'
+  })
+  .option('runs', {
+    'describe': 'Does n runs (eg. 3, 5), and reports the median run\'s numbers.',
+    'type': 'number',
+    'default': 1,
+  })
+  .option('expectations', {
+    'describe': 'Expectations from metrics results. Useful for CI. Config required.',
+    'type': 'boolean',
+    'default': false
+  })
+  .option('submit', {
+    'describe': 'Submit metric results into google sheets. Config required.',
+    'type': 'boolean',
+    'default': false
+  })
+  .option('config', {
+    'describe': 'Path to json config file',
+    'coerce': (arg) => getConfigFromFile(arg),
+    'type': 'string'
+  })
+  .option('disable-cpu-throttling', {
+    'describe': 'Disable CPU throttling',
+    'type': 'boolean',
+    'default': false
+  })
+  .epilogue('For more Lighthouse CLI options see https://github.com/GoogleChrome/lighthouse/#lighthouse-cli-options')
   .check(argv => {
-    if (argv._.length === 0)
-      throw new Error(getMessageWithPrefix('ERROR', 'NO_URL'));
+    //test possible url locations, first from cmd line then from config file
+    if (argv._.length !== 0 || (argv.config !== undefined && argv.config.url))
+      return true;
 
-    return true;
-  })
-  .argv;
+    throw new Error(getMessageWithPrefix('ERROR', 'NO_URL'));
+  }).argv;
 
-const config = Object.assign({}, {flags: cliFlags});
-const fileConfig = getConfigFromFile(cliFlags.config);
-const url = fileConfig.url || cliFlags._[0];
+//Get all explicitly terminal set options, does not include url because url has no default option
+const terminalOptions = Object.keys(yargs.getOptions().default).reduce((accum, prop) => {
+  if (cliFlags[prop] !== yargs.getOptions().default[prop])
+    accum[prop] = cliFlags[prop];
+  return accum;
+}, {});
 
-// get only allowed properties from file config
-Object.keys(fileConfig).forEach(configPropName => {
-  switch(configPropName) {
-    case 'expectations':
-    case 'sheets':
-      config[configPropName] = fileConfig[configPropName];
-      break;
-  }
-});
+//Merge options from all sources. Order indicates precedence (last => most important)
+const options = mergeDeep(yargs.getOptions().default, cliFlags.config, terminalOptions);
```

Comment:
I was thinking about this earlier today. 

My original idea was to be able in the config file to overwrite any option. Url, flags.expectations, json (the bool) and others, not only the expectations object. 

The reason behind it being easy of use. Simply define all options in a file and be done with it.

Otherwise you have to define expectations (bool) in the bash and the actual expectations values in the file (object).

Thoughts? If it's too radical a change I can revert.